app/src/main/java/com/example/bookwormadventuresdeluxe2/CreateAccountActivity.java

```diff
@@ -165,12 +174,10 @@ private void onBackClick(View view)
      * Source: https://stackoverflow.com/questions/48570270/firestore-query-checking-if-username-already-exists
      *
      * @param username Username requiring availability check
+     * @param result Result of query callback
      */
-    public void checkUsernameAvailability(String email, String password, final String username, String phoneNumber)
+    public void checkUsernameAvailability(String username, FirebaseUserGetSet.UsernameTakenCallback result)
```

Comment:
is this call back necessary since ur using a global variable "usernameInUse". the checkUserNameAvailabiliity method can directly set the variable as true or false.

Reply:
yep. it skips past and doesnt finish on time if we only rely on the global variable